vslib/src/sai_vs_debug_counter.cpp

```diff
@@ -1,7 +1,81 @@
 #include "sai_vs.h"
 #include "sai_vs_internal.h"
+#include "sai_vs_state.h"
+#include <unordered_set>
 
-VS_GENERIC_QUAD(DEBUG_COUNTER,debug_counter);
+static std::unordered_set<uint32_t> indices;
+
+static uint32_t get_index()
+{
+    for (uint32_t i = 0; i < 32; i++)
+    {
+        if (indices.find(i) == indices.end())
+        {
+            indices.insert(i);
+            return i;
+        }
+    }
+
+    return UINT32_MAX;
+}
+
+// VS_GENERIC_QUAD(DEBUG_COUNTER,debug_counter);
+sai_status_t vs_create_debug_counter(
+            _Out_ sai_object_id_t *debug_counter_id,
+            _In_ sai_object_id_t switch_id,
+            _In_ uint32_t attr_count,
+            _In_ const sai_attribute_t *attr_list)
+{
+    MUTEX();
+    SWSS_LOG_ENTER();
+
+    /* create debug counter */
+    CHECK_STATUS(meta_sai_create_oid(
+                (sai_object_type_t)SAI_OBJECT_TYPE_DEBUG_COUNTER,
+                debug_counter_id,
+                switch_id,
+                attr_count,
+                attr_list,
+                &vs_generic_create));
+
+    sai_attribute_t attr;
+    attr.id = SAI_DEBUG_COUNTER_ATTR_INDEX;
```

Comment:
why this is actually needed ?

Reply:
The virtual switch was complaining about the read only attribute not being implemented. I'll see if I can find the exact error message.

Reply:
I was finally able to re-test this and it looks like as long as the case is handled in refresh_read_only then we don't actually need to include this for the virtual switch tests to work. I'll go ahead and remove this part.